src/utils_db_query.c

```diff
@@ -1011,7 +1012,13 @@ int udb_query_prepare_result (udb_query_t const *q, /* {{{ */
   if ((q == NULL) || (prep_area == NULL))
     return (-EINVAL);
 
-  udb_query_finish_result (q, prep_area);
```

Comment:
Me strongly disagree too.

>  It's totally fine for the caller not to care about this.

No, that is abnormal.

Before proposed PR, the 'oracle' plugin has no udb_query_finish_result() in o_read_database_query() while udb_query_prepare_result() was there. If no o_read_database_query() called more, preparation area will never be 'clean' and memory will not be freed.

After this patch, rules become more strong, and that is good. Currently, udb_query_prepare_result() and udb_query_finish_result() are done on the same level in 'oracle' plugin, so 'preparation area' freed right after use.

Freeing memory not after use but right before next use is ugly.

Currently, if dig deeper, as 'oracle.c' does not call udb_query_finish_result() and udb_query_delete_preparation_area() also does not call udb_result_finish_result(),   sfree (result_prep_area->metadata_pos) and sfree (result_prep_area->metadata_buffer) are lost on plugin shutdown.

It's totally abnormal for the caller to call udb_query_prepare_result() and do not call udb_query_finish_result().

Rules must be strong and other plugins has correct implementation before this PR:

dbi.c            has udb_query_finish_result() in cdbi_read_database_query()
postgresql.c has udb_query_finish_result() in c_psql_exec_query().

I think what all plugins should behave in the same way and must have udb_query_finish_result() in their code.